testhelpers/steps.go

```diff
@@ -233,6 +235,14 @@ func (ctx *TestContext) IAmUserWithID(id int64) error { // nolint
 	return nil
 }
 
+func (ctx *TestContext) TimeNow(timeStr string) error { // nolint
+	testTime, err := time.Parse(time.RFC3339Nano, timeStr)
+	if err == nil {
+		monkey.Patch(time.Now, func() time.Time { return testTime })
```

Comment:
Instead of patching the time, you should extract your logic so that the time is an input parameter and unit-test that. When it comes to the end-to-end tests, you should not check the time, because using solutions like this you are risking that you are altering the behaviour of the logic under test.

Reply:
It is a discussion on StackOverflow about abused dependency injection pattern (abused DI pattern)... and with time in Go, you need very "constructor" to start passing the time from the app to every component and service. If you have a smart way, I'm interested in seeing it

Reply:
As I wrote, my suggestion would be to instead of calling a function that uses `time.Now()` inside, you provide a `time.Time` value as an input argument. Not sure why you need to check the whole row in the end-to-end tests. Those should be only to make sure the high-level, business functionality works, not the low-level implementation details such as database structure or data stored is a particular row.